pyboy/core/cpu.py

```diff
@@ -168,35 +89,107 @@ def load_state(self, f, state_version):
         self.interrupt_master_enable = f.read()
         self.halted = f.read()
         self.stopped = f.read()
-        logger.debug(
-            f"State loaded: A:{self.A:02x}, F:{self.F:02x}, B:{self.B:02x}, C:{self.C:02x}, D:{self.D:02x}, E:{self.E:02x}, HL:{self.HL:02x}, SP:{self.SP:02x}, PC:{self.PC:02x}, IME:{self.interrupt_master_enable}, halted:{self.halted}, stopped:{self.stopped}"
+        if state_version >= 5:
+            # Interrupt register moved from RAM to CPU
+            self.interrupts_enabled_register = f.read()
+        logger.debug("State loaded: " + self.dump_state())
+
+    def dump_state(self):
+        return (
+            f"A:{self.A:02x}, F:{self.F:02x}, B:{self.B:02x}, C:{self.C:02x}, D:{self.D:02x}, E:{self.E:02x}, "
+            f"HL:{self.HL:02x}, SP:{self.SP:02x}, PC:{self.PC:02x}, "
+            f"Opcode:{self.mb.getitem(self.PC):02x}, "
+            f"Opcode+1:{self.mb.getitem((self.PC+1) & 0xFFFF):02x}, "
+            f"Opcode+2:{self.mb.getitem((self.PC+2) & 0xFFFF):02x}, "
+            f"IME:{self.interrupt_master_enable}, halted:{self.halted}, "
+            f"interrupt_queued:{self.interrupt_queued} "
+            f"stopped:{self.stopped}"
         )
 
+    def set_interruptflag(self, flag):
+        self.interrupts_flag_register |= flag
+
+    def tick(self):
+        if self.check_interrupts():
+            self.halted = False
+            # TODO: We return with the cycles it took to handle the interrupt
+            return 0
+
+        if self.halted and self.interrupt_queued:
+            # GBCPUman.pdf page 20
+            # WARNING: The instruction immediately following the HALT instruction is "skipped" when interrupts are
```

Comment:
Is this implemented here, or just a note? It doesn't seem like it emulates the bug, but hopefully that's not too important to roms that properly avoid it.

Reply:
That's just a quote from the manual. It was actually there already. It's not implemented.

Reply:
Also, I have never heard of a practical example that depends on this.

Reply:
Guess I didn't see it in the old source. I made that comment before I fully understood how the PC was being manipulated (and I still might not get it, based on the other one) so I thought the `+=1` was for the bug.

Reply:
I think the +1 you're look at is just to escape the HALT instruction